handlers/metrics.go

```diff
@@ -80,6 +84,37 @@ func extractMetricsQueryParams(r *http.Request, q *prometheus.MetricsQuery) erro
 		q.ByLabelsOut = lblsout
 	}
 
+	// Get namespace info.
+	namespace, err := k8s.GetNamespace(q.Namespace)
+	if err != nil {
+		return err
+	}
+	// If needed, adjust interval -- Make sure query won't fetch data before the namespace creation
+	intervalStartTime, err := util.GetStartTimeForRateInterval(q.End, q.RateInterval)
+	if err != nil {
+		return err
+	}
+	if intervalStartTime.Before(namespace.CreationTimestamp.Time) {
+		q.RateInterval = fmt.Sprintf("%vs", int(q.End.Sub(namespace.CreationTimestamp.Time).Seconds()))
```

Comment:
Not a big deal (I won't request change for that), but the advantage of using accurate formatting types like "%d" "%s" etc. instead of the catch-all solution %v , is that if there's a refactoring later, if by mistake wrong parameters types being changed and mistakenly passing a wrong type, the problem will be caught earlier ; and tools such as govet or golint (can't remember which) would report this issue. So IMO that's a good practice.

Reply:
Ok! I'll change the format types.